core/templates/dev/head/pages/exploration_player/PlayerServices.js

```diff
@@ -320,6 +320,11 @@ oppia.factory('oppiaPlayerService', [
             classificationResult.answerGroupIndex,
             classificationResult.ruleIndex,
             classificationResult.classificationCategorization);
+
+          if (SolutionManagerService.hasSolutionBeenViewed()) {
```

Comment:
Should we send the event at solution-view time, or at answer-submit time? I had assumed the former, actually. It could be interesting to see how many folks view the solution and then "give up"...

Reply:
@seanlip We documented it as solution-view time, but then if were to record it at answer-submit time, it makes a lot more sense to check if the solution has been viewed before he answered. Is that a stat worth knowing - how many folks give up after seeing the solution. Normal people wouldn't do that right?

Reply:
I wouldn't assume that. The solution may be too hard to type, or the user may not know how to enter it, for example. We've seen things like that.

I'm not sure I understand the first part of your comment, too. I had assumed we would send the event at solution-view time, but I think your code is doing the latter.